### PR TITLE
[Benchmark] upload data to BigQuery directly

### PR DESCRIPTION
**Pull request by SaveTheRbtz**
This diff removes the dependency on `bq` binary and uploads tests results directly to bigquery directly from the benchmark executable.

### CODE REVIEW

File: integration/benchmark/cmd/ci/main.go [hunk 1]
```diff
@@ -216,35 +223,23 @@ func calculateTpsSlices(start, end time.Time, leadTime, sliceTime, commit, goVer
...
 }
 
-func prepareDataForBigQuery(slices []dataSlice, ci bool) {
-	jsonText, err := json.Marshal(slices)
+func sendDataToBigQuery(
+	ctx context.Context,
+	projectName, datasetName, tableName string,
+	slices []dataSlice,
+) error {
+	bqClient, err := bigquery.NewClient(ctx, projectName)
```
Comment (pattyshack) [inline integration/benchmark/cmd/ci/main.go:231]: i'm not familiar with the bigquery api.  do you need some sort of close / cleanup for the client / table?

**Review by zhangchiqing — APPROVED** [review]

File: .github/workflows/performance-ci.yml [hunk 2]
```diff
@@ -16,8 +16,6 @@ on:
       - 'v[0-9]+.[0-9]+'
 
 env:
-  BIGQUERY_DATASET: production_src_flow_tps_metrics
```
Comment (AndrewM-SDET) [inline .github/workflows/performance-ci.yml:19]: This entire file can be removed, this yml now lives in dapperlabs/flow-go to make use of a private self-hosted runner. These changes are already implemented there.

File: integration/benchmark/cmd/ci/main.go [hunk 3]
```diff
@@ -34,18 +35,17 @@ type LoadCase struct {
...
-	ProStartTransaction float64
-	ProEndTransaction   float64
+	GoVersion           string    `bigquery:"goVersion"`
+	OsVersion           string    `bigquery:"osVersion"`
+	GitSha              string    `bigquery:"gitSha"`
+	StartTime           time.Time `bigquery:"startTime"`
+	EndTime             time.Time `bigquery:"endTime"`
+	InputTps            float64   `bigquery:"inputTps"`
+	OutputTps           float64   `bigquery:"outputTps"`
+	ProStartTransaction float64   `bigquery:"proStartTransaction"`
```
Comment (AndrewM-SDET) [inline integration/benchmark/cmd/ci/main.go:47]: These will generate some conflicts when https://github.com/onflow/flow-go/pull/3163 is merged, I will push a fix for this.

File: integration/benchmark/cmd/ci/main.go [hunk 4]
```diff
@@ -71,7 +71,10 @@ func main() {
 	// CI relevant flags
 	tpsFlag := flag.String("tps", "300", "transactions per second (TPS) to send, accepts a comma separated list of values if used in conjunction with `tps-durations`")
 	tpsDurationsFlag := flag.String("tps-durations", "10m", "duration that each load test will run, accepts a comma separted list that will be applied to multiple values of the `tps` flag (defaults to infinite if not provided, meaning only the first tps case will be tested; additional values will be ignored)")
-	ciFlag := flag.Bool("ci-run", false, "whether or not the run is part of CI")
+	_ = flag.Bool("ci-run", false, "(deprecated)")
```
Comment (AndrewM-SDET) [inline integration/benchmark/cmd/ci/main.go:74]: This is only referenced in integration/localhost/Makefile, we should remove it both in that file and here entirely.

**Review by AndrewM-SDET — CHANGES_REQUESTED** [review]
A couple of changes, some I will implement myself after the merge of https://github.com/onflow/flow-go/pull/3163 but all functional work looks good.

**Comment by SaveTheRbtz** [comment]
bors merge

**Pull request merged**